### DISCUSSION TITLE
It is not possible to rewrite cacheTime when using queryClient.setQueryData

### DISCUSSION THREAD

**Discussion by joseucarvajal** [TanStack/query#2146] [Q&A]
I'm iterating through a result array and using setQueryData to memoize individual results. I also have a specific hook to fetch these results individually with userQuery(...). I've tried to set the cacheTime other than 5 mins and it's been not possible when using setQueryData. I've tried many different approaches:

1. set cacheTime as configuration in queryClient.setQueryData(['task', id], data, {cacheTime:5000})
2. set cacheTime within the specific hook with the userQuery(['task', id], fetchFn, {cacheTime:5000})
3. queryClient.setQueryDefaults(['task', id], data, {cacheTime:5000})

None of them have worked, cacheTime always is 300000 when using querClient.setQueryData.

I'm working with react-query v3.13.10.

**Comment by TkDodo** [comment]
Not sure what issue you are trying to solve, what’s your use case? setQueryData can set the query data, not query options. 

However, number 2 in your list should work. useQuery accepts a cacheTime option: https://react-query.tanstack.com/reference/useQuery

**Comment by joseucarvajal** [comment]
Thanks @TkDodo,

The issue I’m facing, is that I haven’t been able to set cacheTime when using setQueryData(...).  This is my question: Is there any way to set cacheTime when using useQueryData?.

**Comment by TkDodo** [comment]
if that's the question then the answer is: no, it's not possible.

but I don't think that is the question. You're clearly trying to solve a problem by setting cacheTime via setQueryData, and I would like to know what that use-case is. Maybe there is a completely different way of solving that problem that doesn't have anything to do with setting cacheTime via setQueryData. But you need to describe your problem in order for me to help out here :)

**Reply by thongtv-1835** [reply]
@TkDodo I have another issue, but it's related to this. I try to save some data in `queryClient` with `queryClient .setQueryData` and I want to persist it as redux (no cache time). Can I do this with `react-query` or I have to create a context for this case. Thank you. Love your lib ❤️

**Reply by TkDodo** [reply]
can you elaborate what the use-case behind that is please? What's the _real issue_ you are trying to solve?

**Reply by thongtv-1835** [reply]
@TkDodo I try to apply `react-query` in my company project, just for learning how to use it. 
We have an event list on home page. And when the user choose a specific event, all other data of other screens will change depend on it. And when the user choose an event, I will save that id in `queryClient` (cache time = Infinity). Is it possible?

**Reply by TkDodo** [reply]
well choosing an event by the user looks like _client_ state to me - owned by the user, not by a backend. So it seems like you want to store that in some client-side state manager

**Reply by thongtv-1835** [reply]
@TkDodo Yeah I got it. Thank you 👍

**Reply by thongtv-1835** [reply]
@TkDodo  But I think It would be useful if we have options for `queryClient .setQueryData` like `useQuery`. Do you plan to support that in the future?

**Reply by TkDodo** [reply]
`setQueryData` only sets `data`, I don't see why it should set anything else 🤔

**Comment by joseucarvajal** [comment]
It would be awesome if serQueryData had the option to set data duration on
cache memory.

**Reply by TkDodo** [reply]
you can do that with `queryClient.setQueryDefaults` for your specific key, or via `queryClient.setDefaultOptions` for all queries
